src/Umbraco.Web.UI.Client/src/views/prevalueeditors/treesource.controller.js

```diff
@@ -118,13 +168,69 @@ angular.module('umbraco')
       });
 		}
 
+    // Dynamic Root specific
+    $scope.$watch("model.value.dynamicRoot", function (newVal, oldVal) {
+
+      const alias = newVal ? newVal.originAlias : null;
+      const originKey = newVal ? newVal.originKey : null;
+
+      if (!alias) {
+        vm.dynamicRootOrigin = null;
+        return;
+      }
+
+      const icon = getIconForOriginAlias(alias);
+      const key = `dynamicRoot_origin${alias}Title`;
+
+      vm.dynamicRootOrigin = {
+        alias: alias,
+        icon: icon
+      };
+
+      setSortingState(newVal.querySteps);
+
+      if (originKey) {
+        const lookupId = udiService.build($scope.model.value.type === 'content' ? 'document' : $scope.model.value.type, originKey);
+
+        vm.dynamicRootOrigin.description = "Loading...";
+        //vm.dynamicRootOrigin.description = $filter('ncNodeName')(lookupId);
```

Comment:
For some reason `ncNodeName` doesn't work here and resolve the value, so the description text was only saying `Loading...`.